### PR TITLE
refactor(www): Move out pageWithSidebar from layout

### PR DESCRIPTION
**Pull request by tesseralis**
## Description

Move out `<PageWithSidebar>` from the main `<Layout>` component and into the specific pages/templates that actually need it -- the docs template and the various feature pages. This further simplifies the scope of the layout component. 

## Related PRs

Depends on: #20763

### CODE REVIEW

File: www/src/templates/template-docs-markdown.js [hunk 1]
```diff
@@ -44,87 +45,88 @@ function DocsTemplate({ data, location, pageContext: { next, prev } }) {
...
-                display: `flex`,
-                alignItems: `flex-start`,
-              },
-            }}
-          >
-            {toc && (
+      <Layout location={location} locale={page.fields.locale}>
+        <PageWithSidebar
+          location={location}
+          enableScrollSync={urlSegment === "docs" ? false : true}
```
Comment (tesseralis) [inline www/src/templates/template-docs-markdown.js]: only the tutorial actually uses this, so we could change this to `urlSegment === "tutorial"` possibly

File: www/src/pages/features/jamstack.js [hunk 2]
```diff
@@ -38,58 +39,60 @@ const JamstackFeaturesPage = ({ data, location }) => {
...
-          <Breadcrumb location={location} />
-          <FeaturesHeader />
-          <h3>Comparison</h3>
-          <p>
-            To see a filtered view of Gatsby with specific JAMstack
-            technologies, choose the technologies to compare and then press
-            Compare:
-          </p>
+    <Layout location={location}>
+      <PageWithSidebar location={location} enableScrollSync={true}>
```
Comment (tesseralis) [inline www/src/pages/features/jamstack.js]: I don't know why `enableScrollSync` is set to true. This is the property that updates the sidebar active item when you scroll through the page, and is only used in the tutorial.

Comment (pieh) [reply www/src/pages/features/jamstack.js]: Let's remove that - it feels like copy & paste case and sidebar in features don't have headings items just pages

File: www/src/templates/template-docs-markdown.js (hunk 1 above)
Comment (pieh) [reply www/src/templates/template-docs-markdown.js]: What sections make use of that template? `docs`, `tutorial` and `contributing`? If so I agree with changing it explicitly to `tutorial`

**Pull request merged**